pygments/formatters/img.py

```diff
@@ -219,14 +227,42 @@ def get_font(self, bold, oblique):
         Get the font based on bold and italic flags.
         """
         if bold and oblique:
+            if self.variable:
+                return self.get_style('BOLDITALIC')
+
             return self.fonts['BOLDITALIC']
         elif bold:
+            if self.variable:
+                return self.get_style('BOLD')
+
             return self.fonts['BOLD']
         elif oblique:
+            if self.variable:
+                return self.get_style('ITALIC')
+
             return self.fonts['ITALIC']
         else:
+            if self.variable:
+                return self.get_style('NORMAL')
+
             return self.fonts['NORMAL']
 
+    def get_style(self, style):
+        """
+        Get the specified style of the font if it is a variable font.
+        If not found, return the normal font.
+        """
+        font = self.fonts[style]
+        for style_name in STYLES[style]:
+            try:
+                font.set_variation_by_name(style_name)
+                return font
+            except ValueError:
+                pass
+            except OSError:
```

Comment:
This indicates an invalid font, right? Shouldn't there be a warning or an error?